AutoRest/Generators/Ruby/Azure.Ruby/AzureRubyCodeGenerator.cs

```diff
@@ -58,6 +66,78 @@ public override void NormalizeClientModel(ServiceClient serviceClient)
             AzureExtensions.NormalizeAzureClientModel(serviceClient, Settings, CodeNamer);
             CorrectFilterParameters(serviceClient);
             base.NormalizeClientModel(serviceClient);
+            AddRubyPageableMethod(serviceClient);
+            ApplyPagination(serviceClient);
+
+        }
+
+        /// <summary>
+        /// Adds method to use for autopagination
+        /// </summary>
+        /// <param name="serviceClient">The service client.</param>
+        private void AddRubyPageableMethod(ServiceClient serviceClient)
+        {
+            if (serviceClient == null)
+            {
+                throw new ArgumentNullException("serviceClient");
+            }
+
+            for (int i = 0; i < serviceClient.Methods.Count; i++)
+            {
+                var method = serviceClient.Methods[i];
+                if (method.Extensions.ContainsKey(AzureExtensions.PageableExtension))
+                {
+                    var pageableExtension = JsonConvert.DeserializeObject<Model.PageableExtension>(method.Extensions[AzureExtensions.PageableExtension].ToString());
+                    if (!method.Extensions.ContainsKey("nextLinkMethod") && !string.IsNullOrWhiteSpace(pageableExtension.NextLinkName)) 
+                    {
+                        serviceClient.Methods.Insert(i, (Method)method.Clone());
+                        method.Name = CodeNamer.GetMethodName(method.Name);
+                        i++;
+                    }
+                    serviceClient.Methods[i].Extensions.Remove(AzureExtensions.PageableExtension);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Changes paginated method signatures to return Page type.
+        /// </summary>
+        /// <param name="serviceClient"></param>
+        private void ApplyPagination(ServiceClient serviceClient)
+        {
+            if (serviceClient == null)
+            {
+                throw new ArgumentNullException("serviceClient");
+            }
+
+            foreach (var method in serviceClient.Methods.Where(m => m.Extensions.ContainsKey(AzureExtensions.PageableExtension)))
+            {
+                string nextLinkName = null;
+                var ext = method.Extensions[AzureExtensions.PageableExtension] as Newtonsoft.Json.Linq.JContainer;
+                if (ext == null)
+                {
+                    continue;
+                }
+                
```

Comment:
minor: Generally i'd do formate document (ctrl + K, ctrl + D) in VS to get rid of unnecessary spaces.